localstack/services/generic_proxy.py

```diff
@@ -186,6 +193,129 @@ def return_response(
         return None
 
 
+class PartitionAdjustingProxyListener(MessageModifyingProxyListener):
+    """
+    Intercepts requests and responses and tries to adjust the partitions in ARNs within the intercepted requests
+    (both in the payload and in the metadata).
+    The partition is selected based on the region in the ARN, or by the default region if the ARN does not contain a
+    region.
+    This listener is used to support other partitions than the default "aws" partition (f.e. aws-us-gov) without
+    rewriting all the cases where the ARN is parsed or constructed within LocalStack or moto.
+    """
+
+    class InvalidRegionException(Exception):
+        """An exception indicating that a region could not be matched to a partition."""
+
+        pass
+
+    arn_regex = re.compile(
+        r"arn:"  # Prefix
+        r"(?P<Partition>(aws|aws-cn|aws-iso|aws-iso-b|aws-us-gov)*):"  # Partition
+        r"(?P<Service>[\w-]*):"  # Service (lambda, s3, ecs,...)
+        r"(?P<Region>[\w-]*):"  # Region (us-east-1, us-gov-west-1,...)
+        r"(?P<AccountID>[\w-]*):"  # AccountID
+        r"(?P<ResourcePath>"  # Combine the resource type and id to the ResourcePath
+        r"((?P<ResourceType>[\w-]*)[:/])?"  # ResourceType (optional, f.e. S3 bucket name)
+        r"(?P<ResourceID>[\w\-/*]*)"  # Resource ID (f.e. file name in S3)
+        r")"
+    )
+
+    def forward_request(
+        self, method: str, path: str, data: MessagePayload, headers: Headers
+    ) -> Optional[RoutingRequest]:
+        return RoutingRequest(
+            method=method,
+            path=self._adjust_partition_in_path(path),
+            data=self._adjust_partition(data),
+            headers=self._adjust_partition(headers),
```

Comment:
On input, we need to change all partitions to `aws`, and on output back to the correct one I think!

Reply:
Fixed.